src/NServiceBus.Transport.Msmq/MsmqUtilities.cs

```diff
@@ -47,28 +47,25 @@ static MsmqAddress GetIndependentAddressForQueue(MessageQueue q)
         public static Dictionary<string, string> ExtractHeaders(Message msmqMessage)
         {
             var headers = DeserializeMessageHeaders(msmqMessage);
-
+            
             //note: we can drop this line when we no longer support interop btw v3 + v4
-            if (msmqMessage.ResponseQueue != null)
+            if (msmqMessage.ResponseQueue != null && headers.ContainsKey(Headers.ReplyToAddress) == false)
```

Comment:
More nitpicking: 

`headers.ContainsKey(Headers.ReplyToAddress) == false`

 should be

`!headers.ContainsKey(Headers.ReplyToAddress)`